src/Compilers/Core/Portable/ConstantValueSpecialized.cs

```diff
@@ -201,6 +210,20 @@
             {
                 return (_value == null) ? "null" : string.Format("\"{0}\"", _value);
             }
+
+            public override string ToString(string? format, IFormatProvider? provider)
+            {
+                int formatLength = RopeValue.Length;
+                if (format is not null && int.TryParse(format, out var len) &&
+                    len >= 3)
```

Comment:
>len >= 3 [](http://example.com/codeflow?start=20&length=8)

We should consider handling `formatLength < 3` as well. We could use `RopeValue.ToString(Math.Max(formatLength - 3, 0))` below.